modules/programs/alacritty.nix

```diff
@@ -37,27 +39,37 @@ in {
         '';
         description = ''
           Configuration written to
-          {file}`$XDG_CONFIG_HOME/alacritty/alacritty.yml`. See
-          <https://github.com/alacritty/alacritty/blob/master/alacritty.yml>
-          for the default configuration.
+          {file}`$XDG_CONFIG_HOME/alacritty/alacritty.yml` or
+          {file}`$XDG_CONFIG_HOME/alacritty/alacritty.toml`
+          (the latter being used for alacritty 0.13 and later).
+          See <https://github.com/alacritty/alacritty/tree/master#configuration>
+          for more info.
         '';
       };
     };
   };
 
-  config = mkMerge [
-    (mkIf cfg.enable {
-      home.packages = [ cfg.package ];
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
 
-      xdg.configFile."alacritty/alacritty.yml" = mkIf (cfg.settings != { }) {
-        # TODO: Replace by the generate function but need to figure out how to
-        # handle the escaping first.
-        #
-        # source = yamlFormat.generate "alacritty.yml" cfg.settings;
-
-        text =
-          replaceStrings [ "\\\\" ] [ "\\" ] (builtins.toJSON cfg.settings);
-      };
-    })
-  ];
+    xdg.configFile."alacritty/${configFileName}" =
+      lib.mkIf (cfg.settings != { }) (lib.mkMerge [
+        (lib.mkIf useToml {
+          source =
+            (tomlFormat.generate configFileName cfg.settings).overrideAttrs
+            (finalAttrs: prevAttrs: {
+              buildCommand = lib.concatStringsSep "\n" [
+                prevAttrs.buildCommand
```

Comment:
(This could have used `postBuild`)

Reply:
This uses `runCommand` underneath which sets `buildCommand` instead of `buildPhase` and so AFAIU there's no pre and post build hooks.
If I add this to `postBuild`, it doesn't do anything.

Reply:
You're right.